controllers/datadogagent/dependencies/store.go

```diff
@@ -275,7 +275,10 @@ func (ds *Store) Cleanup(ctx context.Context, k8sClient client.Client, ddaNs, dd
 
 // GetVersionInfo returns the Kubernetes version
 func (ds *Store) GetVersionInfo() string {
-	return ds.versionInfo.GitVersion
+	if ds.versionInfo != nil {
```

Comment:
just curious, when is this `nil`? would it make sense to add a comment about it?

Reply:
I added that to prevent these and other APM enabling tests from hitting a nil pointer error if there was no version info in the store: [example test](https://github.com/DataDog/datadog-operator/blob/285220df7008ef15c680c0eb772ea6b7561c7575/controllers/datadogagent/controller_test.go#L682). Should that be added as a comment?